hs_tracking/models.py

```diff
@@ -18,31 +18,38 @@
 
 
 class SessionManager(models.Manager):
-    def for_request(self, request):
+    def for_request(self, request, user=None):
+        if hasattr(request, 'user'):
+            user = request.user
+
         signed_id = request.session.get('hs_tracking_id')
         if signed_id:
             tracking_id = signing.loads(signed_id)
             cut_off = datetime.now() - timedelta(seconds=SESSION_TIMEOUT)
             session = None
+
             try:
                 session = Session.objects.filter(
                     variable__timestamp__gte=cut_off).filter(id=tracking_id['id']).first()
             except Session.DoesNotExist:
                 pass
+
             if session is not None:
-                if session.visitor.user is None and request.user.is_authenticated():
+                if session.visitor.user is None and user.is_authenticated():
```

Comment:
No, in the case of an unlogged in user, request.user will be an `AnonymousUser`, which will have the `is_authenticated` method defined (as trivially false).  This above case is here to deal with the fact that my new signal handlers seem to get ahold of the request before the authentication middleware has a chance to add the user to it (if I remember correctly).  So for logins and logouts, I have to pass in the user object to this method directly.